### PR TITLE
Configure whether the network is listed in Packet Broker

### LINKED ISSUE #4447: Configure listed explicitly on Packet Broker registration
#### Summary

Configure whether a network is listed in Packet Broker explicitly in the Console and CLI.

#### Why do we need this?

Networks can only discover one another for peering configuration if their network is publicly listed in Packet Broker.

The reason why listed exists is to avoid that private networks get exposed automatically, leading to potential privacy issues.

So, this is a manual action. In TTSOS, this works with the `pba.registration` configuration. In TTSE, this references https://github.com/TheThingsIndustries/lorawan-stack/issues/2616.

#### What is already there? What do you see now?

There should be a slider to control whether the network is listed.

#### What is missing? What do you want to see?

Just like there is a slider for registration, there should be a slider for whether the network is listed. This should show the value from Packet Broker, and a change should update the registration in Packet Broker. The default value when registering at Packet Broker should be taken from the current configuration, like now.

#### Environment

CLI + Console

#### How do you propose to implement this?

Add new API to control whether the network is listed.

#### How do you propose to test this?

🐒 

#### Can you do this yourself and submit a Pull Request?

Yes, but probably not the Console part. Will reassign when necessary.

### LINKED ISSUE #4499: Remove deprecated TLS authentication
#### Summary

Remove deprecated TLS authentication

#### Why do we need this?

Packet Broker doesn't support it anymore.

It was deprecated since 3.10.6.

#### What is already there? What do you see now?

TLS authentication in Packet Broker Agent.

#### What is missing? What do you want to see?

Have it removed entirely.

#### How do you propose to implement this?

- Remove from config
- Remove initializing dial options
- Remove TLS in tests

#### How do you propose to test this?

CI and integration testing. Packet Broker doesn't support TLS authentication anymore; it's just testing safely removal of deprecated code and making tests independent of TLS authentication.

#### Can you do this yourself and submit a Pull Request?

Yes

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Closes #4447 
Closes #4499

#### Changes

- Register switch is now disabled if Packet Broker Agent isn't authenticated to alter the registration itself
- Added a new switch to configure whether the network is listed in Packet Broker. This overrides any registration settings, if set
- Clarified some texts
- Packet Broker Agent and API changes to facilitate the above changes
- Removed some technical debt (#4499); this is non-functional

<details>
<summary>Screenshots</summary>

Screenshot 2021-08-07 at 22 00 08

Screenshot 2021-08-07 at 22 01 12

Screenshot 2021-08-07 at 22 00 21

</details>

#### Testing

Local integration testing

##### Regressions

None expected

#### Notes for Reviewers

@kschiffer is the main reviewer for the Console

@adriansmares is the main reviewer for the API and PBA part. Note that removing the TLS authentication is only code cleanup and isn't really functional; it hasn't been supported by Packet Broker for a long time now, it was only used in tests.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by htdvisser — APPROVED** [review]
API changes LGTM. Didn't review the rest (not code-owned).

**Comment by kschiffer** [comment]
I've applied a couple of styling, structural and UX fixes. As such, I'll forward the review to @bafonins to avoid self-review. The changes not altered by me LGTM though.
image

File: pkg/webui/console/views/admin-packet-broker/messages.js [hunk 1]
```diff
@@ -18,19 +18,30 @@ export default defineMessages({
...
-  registerThisNetwork: 'Register this network',
-  network: 'Network: {network}',
-  homeNetworkEnabled: 'Home network enabled',
-  homeNetworkDisabled: 'Home network disabled',
-  forwarderEnabled: 'Forwarder enabled',
-  forwarderDisabled: 'Forwarder disabled',
+  networkVisibility: 'Network visibility',
   packetBrokerRegistrationDesc:
-    "To enable package exchange from or to your home network, it is necessary to register your network. This will make your network known to the Packet Broker and enable you to configure your network's package exchange behavior.",
+    "To enable package exchange from or to your home network, it is necessary to register your network. This will make your network known to Packet Broker and enable you to configure your network's packet exchange behavior.",
```
Comment (johanstokking) [inline pkg/webui/console/views/admin-packet-broker/messages.js]: We're mixing terms:

- Exchange traffic
- Peering
- Package exchange

These are all interchangeable.

To add more terms, we use the terms messages (TTS), frames (used in LoRaWAN spec, lower level), packets (like Packet Broker) and packages.

I think we should start stripping a few terms here, beginning with "package" and "package exchange". Instead, we should go with "peering", "traffic exchange" and "messages".

Comment (kschiffer) [reply pkg/webui/console/views/admin-packet-broker/messages.js]: Sorry, this was an unintentional change actually.

**Comment by johanstokking** [comment]
Cool. Are you fixing CI? You can then go ahead by approving and merging.

**Review by adriansmares — APPROVED** [review]
Code owned files LGTM. Good stuff.

File: pkg/webui/console/views/admin-packet-broker/admin-packet-broker.js [hunk 2]
```diff
@@ -146,40 +230,51 @@ const PacketBroker = ({ match }) => {
...
-                <NotFoundRoute />
-              </RouteSwitch>
-            </RequireRequest>
-          </Col>
+          <>
+            <Col lg={8} md={12}>
+              <Message content={m.networkVisibility} component="h3" className={style.subTitle} />
+              <Row gutterWidth={48}>
+                <Col md={4}>
+                  <label className={classnames(style.toggleContainer)}>
```
Comment (bafonins) [inline pkg/webui/console/views/admin-packet-broker/admin-packet-broker.js]:
```suggestion
                  <label className={style.toggleContainer}>
```

**Review by bafonins — APPROVED** [review]
`webui` lgtm

**Pull request merged**